src/main/cpp/src/CastStringJni.cpp

```diff
@@ -111,4 +118,72 @@ JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_CastStrings_fromDecimal
   }
   CATCH_CAST_EXCEPTION(env, 0);
 }
+
+JNIEXPORT jlong JNICALL Java_com_nvidia_spark_rapids_jni_CastStrings_toIntegersWithBase(
+  JNIEnv* env, jclass, jlong input_column, jint base)
+{
+  JNI_NULL_CHECK(env, input_column, "input column is null", 0);
+  using namespace cudf;
+  try {
+    jni::auto_set_device(env);
+    auto const res_data_type = data_type(type_id::UINT64);
+
+    auto const input_view{*reinterpret_cast<column_view const*>(input_column)};
+    const strings_column_view input_strings{input_view};
+    auto integer_view = [&] {
+      switch (base) {
+        case 10: {
+          // TODO implement it in the kernel
+          auto const regex = strings::regex_program::create(R"(^\s*([0-9]+).*)");
+          auto const dec_str_table = strings::extract(input_strings, *regex);
+          const strings_column_view dec_str_view{dec_str_table->get_column(0)};
+          return strings::to_integers(dec_str_view, res_data_type);
```

Comment:
It doesn't look like we have any overflow checking in here, but Spark does do overflow checking, and just returns a -1 in non-ansi mode. Is the something that will be handled by the plugin, or are we lucky and happen to get the right answer out of CUDF.

Reply:
At this point there is no overflow checking indeed. We currently support only the [happy path](https://github.com/NVIDIA/spark-rapids/pull/8925/files#diff-af93e5faff863709f2c9de84607d8bd79c6cc0e2454b6f982c693c10ad721402R784-R803) avoiding the overflow